vllm/config.py

```diff
@@ -251,6 +263,18 @@ def get_sliding_window(self) -> Optional[int]:
             return None
         return getattr(self.hf_text_config, "sliding_window", None)
 
+    def get_sliding_window(self) -> Optional[int]:
```

Comment:
How does it work for the model that already has sliding window like mistral?

Reply:
Im not sure what you mean?

If the user does not specify `--disable-sliding-window` then we use sliding window if the model supports it

Reply:
oh maybe it is a dumb question, but my question is for models that has slinding window by default https://huggingface.co/mistralai/Mistral-7B-v0.1/blob/26bca36bde8333b5d7f72e9ed20ccda6a618af24/config.json#L18, if we use --disable-sliding-window, does it work properly?

Reply:
Yes, specifically what this does is handle a case like Mistral.

`--disable-sliding-window` means we turn off sliding window and set `max_model_len=sliding_window`

So in the case of Mistral, we then would treat the model as a 4096 ctx-len model with no sliding window. 

The reason for this feature is that if we want to use features that are incompatible with sliding window (e.g. APC or chunked prefill), then there is a pathway to disable sliding window

Reply:
I see. that makes sense! Thanks for the explanation